compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/ExternalWarnings/Script.hs

```diff
@@ -0,0 +1,46 @@
+-- Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE PatternSynonyms #-}
+
+-- | For compiler level warnings on Daml.Script
+module DA.Daml.LFConversion.ExternalWarnings.Script (topLevelWarnings) where
+
+import DA.Daml.UtilGHC
+import DA.Daml.LFConversion.ConvertM
+import DA.Daml.LFConversion.Utils
+import "ghc-lib" GhcPlugins as GHC hiding ((<>))
+import "ghc-lib" TyCoRep
+import SdkVersion (sdkPackageVersion)
+
+-- Identifying the Daml.Script module
+damlScriptUnitId :: GHC.UnitId
+damlScriptUnitId = fsToUnitId (fsLit $ "daml-script-" <> sdkPackageVersion)
```

Comment:
the latter, not related to the other thing (running a `daml-script` with a JSON argument).

In particular, note that they wouldn't be able to give `myTestX` a signature, since the old and new `Script` are different types, but without one it's inferred as the old one

Reply:
oh right, using the script function stuff. When they use a json argument, the application actually happens on the scenario service, and is mostly untyped

Reply:
There is a solution for above btw, and its what we do for std-lib i believe. Which is simply checking the prefix, and allowing the "version" to be anything. Its not the most safe thing in the world, if you had a module like daml-script-2, you might get some weird behaviour, but the same is true of if you had `daml-stdlib-2`

Reply:
oh, I forgot this was about polymorphic `Script`s, but consider,

```haskell
module SomeDataDep

import Daml.Script -- SDK 3.5.0

data X = X {...}

mkTest : forall a. X -> Script a
mkTest = (...)
```

```haskell
module MyPackage

import Daml.Script -- SDK 3.6.0
import SomeDataDep (mkTest)

-- [inferred] forall a. (daml-script-3.5.0):Script a
--                       ^^^^^^^^^^^^^^^^^- can't warn about this one
myTest = mkTest (X ...)
```

Reply:
Makes sense, so we'll need to cover all Script versions
Do you think the same approach to daml-stdlib is okay? Checking the prefix?

Reply:
I don't think this is suuuper likely, though, so I would be happy to leave this as a separate issue in our backlog.

Reply:
Have applied the fix, lmk what you think

Reply:
right, the "risk" now is that someone who 1. named their package `daml-script-<something>`, 2. defined a data type `data Script a` and 3. declared an expression with inferred type `forall a. Script a` will get the warning. I don't think that's likely, and even then it's just a warning, so ✅